src/components/ExecuteButton.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from 'react';
+
+import PropTypes from 'prop-types';
+import { downloadPlaybook } from '../api';
+import {
+    Button, Modal, TextContent, Text, TextVariants } from '@patternfly/react-core';
+import { TableHeader, Table, TableBody, TableVariant } from '@patternfly/react-table';
+import { CheckCircleIcon } from '@patternfly/react-icons';
+import { Skeleton } from '@redhat-cloud-services/frontend-components';
+import './ExecuteButton.scss';
+
+const ExecuteButton = ({ isLoading, data, getConnectionStatus, remediationId, issueCount }) => {
+    const [ open, setOpen ] = useState(false);
+    const [ isUserEntitled, setIsUserEntitled ] = useState(false);
+    useEffect(() => {
+        window.insights.chrome.auth.getUser().then(user => setIsUserEntitled(user.entitlements.smart_management.is_entitled));
+    }, []);
+
+    const styledConnectionStatus = (status) => (
+        status === 'connected'
+            ? (<div>
+                <CheckCircleIcon
+                    className="ins-c-remediations-reboot-check-circle ins-c-remediations-connection-status"
+                    aria-label="connection status"/>
+                Ready
+            </div>)
+            : status
+    );
+    const [ connected, disconnected ] = data.reduce(
+        ([ pass, fail ], e) => (e.connection_status === 'connected' ? [ [ ...pass, e ], fail ] : [ pass, [ ...fail, e ] ])
+        , [ [], [] ]
+    );
+
+    const rows = [ ...connected, ...disconnected ].map(con =>
+        ({ cells: [ con.executor_name || 'Direct connection', con.system_count, { title: styledConnectionStatus(con.connection_status) }]})
+    );
+    const systemCount = connected.reduce((acc, e) => e.system_count + acc, 0);
+
+    const pluralize = (number, str) => number > 1 ? `number ${str}s` : `${number} ${str}`;
```

Comment:
There is typo in the first string literal, instead of `number` wrap it in eval `${number} ${str}s`